pymongo/change_stream.py

```diff
@@ -101,3 +179,59 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
+
+
+class CollectionChangeStream(ChangeStream):
```

Comment:
I don't think we can change the type returned by Collection.watch. I think that should continue to return ChangeStream. Your abstract class can be an internal type line _ChangeStream.

Reply:
The 'correct' choice probably takes into account how users check the 'type' of what is returned by `watch()`. 

If we use a base class `_ChangeStream` and call the change stream associated with collections `ChangeStream`, `isinstance(coll.watch(), ChangeStream)` will be `True`, but `isinstance(db.watch(), ChangeStream)` and `isinstance(client.watch(), ChangeStream) will be `False` (as they will inherit from `_ChangeStream` base class). 

I think it is much more intuitive to have everything returned by `watch()` methods be of the same (implicit) type (as determined by `isinstance`). I can't think of a non-hacky way to do that while simultaneously ensuring that the explicit type from `coll.watch()` is `ChangeStream`.

With that in mind, how do you want to proceed?

Reply:
That's a good point. isinstance checks will still pass. I take back what I wrote. Just note this in the changelog and I think we'll be OK.

Reply:
Adding the following to the changelog:
```
- The :meth:`pymongo.collection.Collection.watch` method now returns an
  instance of the :class:`~pymongo.change_stream.CollectionChangeStream`
  class which is a subclass of :class:`~pymongo.change_stream.ChangeStream`.
```